src/App.tsx

```diff
@@ -1,9 +1,223 @@
+      setFilterType('phones');
+    } else if (pathname === '/tablets') {
+      setFilterType('tablets');
+    } else if (pathname === '/accessories') {
+      setFilterType('accessories');
+    } else if (pathname === '/favourites') {
+      setFilterType('favourites');
+    } else {
+      setFilterType('');
+    }
+
+    setQuery('');
+    updateSearch({ query: null });
+    history(-1);
+  }, [pathname]);
+
+  return (
+    <div className="app">
+      <Header
+        filterType={filterType}
+        filterQuery={filterQuery}
+        updateSearch={updateSearch}
+      />
+
+      <main style={{ flexGrow: 1 }}>
+        <Routes>
+          <Route
+            path="/"
+            element={(
+              <Home pathname={pathname} />
+            )}
+          />
+
+          <Route
+            path="/home"
+            element={<Navigate to="/" replace />}
+          />
+
+          <Route
+            path="/phones"
+            element={(
+              <>
+                <Navigation
+                  pathname={pathname}
+                />
+                <Products
+                  productType={ProductTypes.phones}
+                  pathname={pathname}
+                  type="phone"
+                  updateSearch={updateSearch}
+                  page={page}
+                  sort={sort}
+                  perPage={perPage}
+                />
+              </>
+            )}
+          />
+
+          <Route
+            path={`/phones/:${activeProduct?.id}`}
+            element={(
+              <>
+                <Navigation
+                  pathname={pathname}
+                />
+                <ProductDetails pathname={pathname} />
+              </>
+            )}
+          />
+
+          <Route
+            path="/tablets"
+            element={(
+              <>
+                <Navigation
+                  pathname={pathname}
+                />
+                <Products
+                  productType={ProductTypes.tablets}
+                  pathname={pathname}
+                  type="tablet"
+                  updateSearch={updateSearch}
+                  page={page}
+                  sort={sort}
+                  perPage={perPage}
+                />
+              </>
+            )}
+          />
+
+          <Route
+            path={`/tablets/:${activeProduct?.id}`}
+            element={(
+              <>
+                <Navigation
+                  pathname={pathname}
+                />
+                <ProductDetails pathname={pathname} />
+              </>
+            )}
+          />
+
+          <Route
+            path="/accessories"
+            element={(
+              <>
+                <Navigation
+                  pathname={pathname}
+                />
+                <Products
+                  productType={ProductTypes.accessories}
+                  pathname={pathname}
+                  type="accessory"
+                  updateSearch={updateSearch}
```

Comment:
when you create a custom hook, this prop will be redundant because the usage can simply use this hook in the Products component